tests/unit/test_okta_plugin.py

```diff
@@ -169,6 +169,27 @@ def test_no_cached_token(mocker, mock_plugin_service, mock_session, mock_func, m
     assert WrapperProperties.USER.get(test_props) == _DB_USER
     assert WrapperProperties.PASSWORD.get(test_props) == _TEST_TOKEN
 
+@patch("aws_advanced_python_wrapper.okta_plugin.OktaAuthPlugin._token_cache", _token_cache)
+def test_no_cached_token_raises_exception(mocker, mock_plugin_service, mock_session, mock_func, mock_client, mock_dialect, mock_credentials_provider_factory):
+    test_props: Properties = Properties({"plugins": "okta", "user": "postgresqlUser", "idp_username": "user", "idp_password": "password"})
+
+    exception_message = "generic exception"
+    mock_func.side_effect = Exception(exception_message)
+
+    target_plugin: OktaAuthPlugin = OktaAuthPlugin(mock_plugin_service, mock_credentials_provider_factory, mock_session)
+
+    with pytest.raises(Exception) as e_info:
+        target_plugin.connect(
+            target_driver_func=mocker.MagicMock(),
+            driver_dialect=mock_dialect,
+            host_info=_PG_HOST_INFO,
+            props=test_props,
+            is_initial_connection=False,
+            connect_func=mock_func)
+
+    assert e_info.type == Exception
```

Comment:
```suggestion
    mock_client.generate_db_auth_token.assert_called_with(
        DBHostname="pg.testdb.us-east-2.rds.amazonaws.com",
        Port=5432,
        DBUsername="postgresqlUser"
    )
    
    assert e_info.type == Exception
```

Reply:
Since this test doesn't have any token in the cache, lets also assert that `generate_db_auth_token` has been called to generate a fresh token.

Same suggestions for okta.
`DB_USER` is a required connection param for federated auth plugins.